include/deal.II/base/polynomial_space.h

```diff
@@ -141,7 +141,9 @@ class PolynomialSpace
   void compute (const Point<dim>            &unit_point,
                 std::vector<double>         &values,
                 std::vector<Tensor<1,dim> > &grads,
-                std::vector<Tensor<2,dim> > &grad_grads) const;
+                std::vector<Tensor<2,dim> > &grad_grads = Utilities::EmptyVectorGenerator<Tensor<2,dim> >().get_reference(),
+                std::vector<Tensor<3,dim> > &grad_hessians = Utilities::EmptyVectorGenerator<Tensor<3,dim> >().get_reference(),
+                std::vector<Tensor<4,dim> > &grad_hessian_grads = Utilities::EmptyVectorGenerator<Tensor<4,dim> >().get_reference()) const;
```

Comment:
Within the library, it is possible to change every call of compute() to accept five parameters (value and derivatives up to fourth), and initialize the empty vectors before the call. This would break any user code that calls the public compute() functions, but then again it seems like a rarely called function outside of the library to me.

On the other hand, since an empty vector being passed to these functions is a signal that this empty vector should not be touched, do the lifetime considerations still come into play?